src/libraries/System.Runtime/tests/System/StringTests.cs

```diff
@@ -149,6 +149,22 @@ public static void Create_ReturnsExpectedString(string expected)
             Assert.Equal(expected, result);
         }
 
+        [Fact]
+        public static void Create_InterpolatedString_ConstructsStringAndClearsBuilder()
+        {
+            Span<char> initialBuffer = stackalloc char[16];
+
+            InterpolatedStringBuilder builder = InterpolatedStringBuilder.Create(0, 0, CultureInfo.InvariantCulture, initialBuffer);
+            builder.AppendLiteral("hello");
+            Assert.Equal("hello", string.Create(CultureInfo.InvariantCulture, initialBuffer, ref builder));
+            Assert.Equal("", string.Create(CultureInfo.InvariantCulture, initialBuffer, ref builder));
```

Comment:
Do we already have a test that writes more than initialBuffer.Length?

Reply:
https://github.com/dotnet/runtime/blob/516db737cd41e27830698e538b36d12d70172630/src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/InterpolatedStringBuilderTests.cs#L422-L439